File: 6zigZagConversion/zigZagConversion.js
/**
 * @param {string} s
 * @param {number} numRows
 * @return {string}
 */
var convert = function(s, numRows) {
    if (Object.is(numRows, 1)) return s
    else
    {
        const result = Array(numRows).fill('')
        let row = 0, direction = 1
        for (const _ of s)
        {
            result[row] += _
            if (Object.is(row, 0)) direction = 1
            else if (Object.is(row, numRows - 1)) direction = -1
            row += direction
        }
        return result.join('')
    }
};

File: 6zigZagConversion/zigZagConversion.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        if (numRows == 1) return s;
        else
        {
            std::vector<std::string> result(numRows);
            auto row{0}, direction{1};
            for (auto const _: s)
            {
                result.at(row) += _;
                if (row == 0) direction = 1;
                else if (row == numRows - 1) direction = -1;
                row += direction;
            }
            std::stringstream out;
            std::copy(std::cbegin(result), std::cend(result), std::ostream_iterator<std::string>{out});
            return out.str();
        }
    }
};
